src/ts/content/extensions/util/contentWarningCheck.ts

```diff
@@ -0,0 +1,82 @@
+import {OffSave, OnSave} from "../../entities/bookForm";
+import {loaderOverlaid} from "../../../common/ui/loadingIndicator";
+import {getTagsFromElement, getTagTrees, TagTrees} from "../../adapters/tags";
+import {createModal} from "../../../common/ui/modal";
+import {UIColour} from "../../../common/ui/colour";
+
+const contentWarningIsPresent = (commentsTextArea: HTMLTextAreaElement): boolean =>
+	commentsTextArea.textContent.split("\n").some((line) => line.startsWith("CW:"));
+
+const saveContentWarning = (commentsTextArea: HTMLTextAreaElement, contentWarning: string): void => {
+	const cleanedWarning = contentWarning.replace(/^CW:\s*/, "");
+	commentsTextArea.textContent = `CW: ${cleanedWarning}\n${commentsTextArea.textContent}`;
+	commentsTextArea.dispatchEvent(new Event("change"));
+};
+
+const handleContentWarning = async (
+	tagsRequiringWarning: string[],
+	commentsTextArea: HTMLTextAreaElement
+): Promise<boolean> => {
+	if (tagsRequiringWarning.length > 0) {
+		return new Promise<boolean>((resolve) =>
+			createModal({
+				text: "Don't forget a Content Warning. The following tags require a Content Warning",
+				subText: tagsRequiringWarning,
+				elements: [
+					{
+						kind: "input",
+						placeholder: "Enter a content warning",
+						ensureNonEmpty: true,
+						text: "Save",
+						colour: UIColour.GREY,
+						onSelect: async (userText) => {
+							saveContentWarning(commentsTextArea, userText);
+							resolve(true);
+						},
+					},
+					{kind: "button", text: "Cancel", colour: UIColour.PURPLE, onClick: async () => resolve(false)},
+				],
+				colour: UIColour.PURPLE,
+			})
+		);
+	} else {
+		return true;
+	}
+};
+const getWarningRequiredTags = (commentsTextArea: HTMLTextAreaElement, tags: string[], trees: TagTrees): string[] => {
+	if (contentWarningIsPresent(commentsTextArea)) {
```

Comment:
does this mean that if there's one `cw:` in the comments box, we don't show the warning?

Reply:
this is reasonable, i'm just wondering

Reply:
Yeah exactly. There just needs to be some content warning